libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1104,29 +1104,60 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	case FunctionType::Kind::ABIEncode:
 	case FunctionType::Kind::ABIEncodePacked:
 	case FunctionType::Kind::ABIEncodeWithSelector:
+	case FunctionType::Kind::ABIEncodeCall:
 	case FunctionType::Kind::ABIEncodeWithSignature:
 	{
 		bool const isPacked = functionType->kind() == FunctionType::Kind::ABIEncodePacked;
 		solAssert(functionType->padArguments() != isPacked, "");
 		bool const hasSelectorOrSignature =
 			functionType->kind() == FunctionType::Kind::ABIEncodeWithSelector ||
+			functionType->kind() == FunctionType::Kind::ABIEncodeCall ||
 			functionType->kind() == FunctionType::Kind::ABIEncodeWithSignature;
 
 		TypePointers argumentTypes;
 		TypePointers targetTypes;
 		vector<string> argumentVars;
-		for (size_t i = 0; i < arguments.size(); ++i)
+		string selector;
+		vector<ASTPointer<Expression const>> argumentsEncodeFunction;
+
+		if (functionType->kind() == FunctionType::Kind::ABIEncodeCall)
 		{
-			// ignore selector
-			if (hasSelectorOrSignature && i == 0)
-				continue;
-			argumentTypes.emplace_back(&type(*arguments[i]));
-			targetTypes.emplace_back(type(*arguments[i]).fullEncodingType(false, true, isPacked));
-			argumentVars += IRVariable(*arguments[i]).stackSlots();
+			solAssert(arguments.size() == 2, "");
+			auto const tuple = dynamic_cast<TupleExpression const*>(arguments[1].get());
+
+			// Account for tuples with one component which become that component
+			if (!tuple)
+				argumentsEncodeFunction.push_back(arguments[1]);
+			else
+				for (auto component: tuple->components())
+					argumentsEncodeFunction.push_back(component);
 		}
+		else
+			for (size_t i = 0; i < arguments.size(); ++i)
+			{
+				// ignore selector
+				if (hasSelectorOrSignature && i == 0)
+					continue;
+				argumentsEncodeFunction.push_back(arguments[i]);
+			}
 
-		string selector;
-		if (functionType->kind() == FunctionType::Kind::ABIEncodeWithSignature)
+		for (auto const& argument: argumentsEncodeFunction)
+		{
+			argumentTypes.emplace_back(&type(*argument));
+			targetTypes.emplace_back(type(*argument).fullEncodingType(false, true, isPacked));
+			argumentVars += IRVariable(*argument).stackSlots();
+		}
+
+		if (functionType->kind() == FunctionType::Kind::ABIEncodeCall)
+		{
+			auto const functionPtr = dynamic_cast<FunctionTypePointer>(&type(*arguments[0]));
+			solAssert(functionPtr, "");
+			if (functionPtr->hasDeclaration())
```

Comment:
I also don't see why we need this here. `IRVariable(*arguments[0])` should handle both cases.

Reply:
It's still not pushed chris, I am still working on cameels comments :)

Reply:
Nevermind, your comment is valid non-the-less

Reply:
In this particular case it probably just saves as a conversation from 32 -> 4